app/components/satis/dropdown/component_controller.js

```diff
@@ -872,40 +893,38 @@ export default class extends ApplicationController {
     return true;
   }
 
-  showSelectedItem(){
-    if(this.isMultipleValue
+  hasFocus() {
+    return this.element.querySelector(":focus") !== null
+  }
+
+  showSelectedItem() {
+    if (this.isMultipleValue
       || this.freeTextValue
       || this.hiddenSelectTarget.options.length === 0
-      || this.element.querySelector(":focus") === null
-  ) return false;
-
-      Array.from(this.hiddenSelectTarget.options).forEach((option) => {
-        let item = this.itemsTarget.querySelector(`[data-satis-dropdown-item-value="${option.value}"]`)
-        if (!item) {
-          item = document.createElement("div")
-          item.innerHTML = `
-            <div class="cursor-pointer w-full dark:border-gray-700 border-gray-100 border-b hover:bg-primary-200">
-              <div class="flex w-full items-center p-2 pl-2 border-transparent border-l-2 hover:border-teal-100">
-                <div class="w-full items-center flex">
-                  <div class="mx-2 -mt-1">${option.text}</div>
-                </div>
-              </div>
-            </div>
-          `;
-          item.setAttribute("data-satis-dropdown-target", "item");
-          item.setAttribute("data-action", "click->satis-dropdown#select");
-          item.setAttribute("data-satis-dropdown-item-value", option.value);
-          item.setAttribute("data-satis-dropdown-item-text", option.text);
-          this.itemsTarget.appendChild(item)
-          this.copyItemAttributes(option, item)
-        }
-        item.classList.toggle("hidden", false)
-        this.selectedIndex = 0
-        this.highLightSelected()
-        if(!this.resultsShown )
-          this.showResultsList()
-      })
-    return true;
+      || !this.hasFocus()
+    ) return false;
+
+    const option = this.hiddenSelectTarget.options[0]
+    let item = this.itemsTarget.querySelector(`[data-satis-dropdown-item-value="${option.value}"]`)
+    if (item) {
+      item.classList.remove("hidden")
+    } else {
+      item = this.selectedItemsTemplateTarget.content.querySelector(`[data-satis-dropdown-item-value="${option.value}"]`)
+      if (item) {
+        item = item.cloneNode(true)
```

Comment:
?

Reply:
Well, indeed, this is could use some explanation. The idea, I believe, is to move the HTML here to the selectedItemsTemplate in the view.
For reviewers it handy to have some context which could be explained in the ticket/PR